### PR TITLE
Add tests take bad arguments 

### PR DESCRIPTION
**Pull request by Lobotuerk**

### CODE REVIEW

File: test_rmw_implementation/test/test_subscription.cpp [hunk 1]
```diff
@@ -377,3 +377,29 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  rmw_ret_t ret;
```
Comment (Blast545) [inline test_rmw_implementation/test/test_subscription.cpp:382]: Suggestion: move variable declaration to line 387 where it is used for the first time

File: test_rmw_implementation/test/test_subscription.cpp [hunk 2]
```diff
@@ -377,3 +377,29 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+  bool taken;
```
Comment (Blast545) [inline test_rmw_implementation/test/test_subscription.cpp:383]: Nit: move closer to the place where it is used and don't leave it uninitialized

**Review by Blast545 — APPROVED** [review]
Minor suggestions, LGTM

**Review by hidmic — COMMENTED** [review]
Same comments as in #126 apply.

File: test_rmw_implementation/test/test_subscription.cpp (hunk 1 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:382]: Updated

File: test_rmw_implementation/test/test_subscription.cpp (hunk 2 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:383]: Updated

File: test_rmw_implementation/test/test_subscription.cpp [hunk 3]
```diff
@@ -377,3 +377,56 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
...
+  EXPECT_EQ(RMW_RET_INCORRECT_RMW_IMPLEMENTATION, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(taken, false);
+  EXPECT_EQ(output_message, test_message);
+  rmw_reset_error();
+  sub->implementation_identifier = implementation_identifier;
+}
+
+bool operator==(const test_msgs__msg__BasicTypes & m1, const test_msgs__msg__BasicTypes & m2)
+{
+  if (m1.bool_value == m2.bool_value &&
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:415]: @Lobotuerk nit: 
```suggestion
  return (m1.bool_value == m2.bool_value && ...);
```
also works.

File: test_rmw_implementation/test/test_subscription.cpp [hunk 4]
```diff
@@ -377,3 +377,56 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  bool taken = false;
+  test_msgs__msg__BasicTypes output_message{};
+  test_msgs__msg__BasicTypes test_message{};
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:384]: @Lobotuerk nit: 
```suggestion
  test_msgs__msg__BasicTypes original_message = output_message;
```

File: test_rmw_implementation/test/test_subscription.cpp [hunk 5]
```diff
@@ -377,3 +377,56 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  bool taken = false;
+  test_msgs__msg__BasicTypes output_message{};
```
Comment (hidmic) [inline test_rmw_implementation/test/test_subscription.cpp:383]: @Lobotuerk nit: it'd be nice to not use the default initialization i.e. pick a couple fields and change them.

**Review by hidmic — APPROVED** [review]
LGTM but for a few nits.

File: test_rmw_implementation/test/test_subscription.cpp (hunk 3 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:415]: Updated

File: test_rmw_implementation/test/test_subscription.cpp (hunk 4 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:384]: Updated

File: test_rmw_implementation/test/test_subscription.cpp (hunk 5 above)
Comment (Lobotuerk) [reply test_rmw_implementation/test/test_subscription.cpp:383]: Updated

**Comment by Lobotuerk** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12400/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7368/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10111/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12313/)

**Comment by Lobotuerk** [comment]
@ros-pull-request-builder retest this please.

**Comment by Lobotuerk** [comment]
@ros-pull-request-builder retest this please.

**Comment by Lobotuerk** [comment]
@ros-pull-request-builder retest this please.

**Comment by Lobotuerk** [comment]
@ros-pull-request-builder retest this please

**Comment by hidmic** [comment]
@Lobotuerk one more conflict to resolve!

**Comment by Lobotuerk** [comment]
CI up to `test_rmw_implementation` after rebase:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12444/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7411/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10154/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12359/)

**Comment by Lobotuerk** [comment]
CI up to test_rmw_implementation after 4b14d93 fixes:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12446/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7413/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10156/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12361/)

**Review by hidmic — APPROVED** [review]

**Comment by Lobotuerk** [comment]
Merging with green CI

**Pull request merged**